include/CppUTest/UtestMacros.h

```diff
@@ -249,8 +249,10 @@
 #define BITS_EQUAL_TEXT(expected, actual, mask, text)\
   BITS_LOCATION(expected, actual, mask, text, __FILE__, __LINE__)
 
+#define ROBUST_SIZEOF(type) (2*sizeof(type)/sizeof(short)) /* For the Cl2000, sizeof(char) == sizeof(short) == 1 */
```

Comment:
I don't remember exactly. There are definitely test failures. There are display errors, e.g. when 1110111100000001 should have been displayed, 00000001 would be displayed, and so on
For the TI, a "byte" is 16 bit, hence sizeof is off compared to other platforms, by a factor of 2.

I think ifdefing testcases is not appropriate here, because when I display a short, the expected result really is 0xABCD and not 0xCD, and so on.

The problem wasn't there before, so I think only the BITS_EQUAL stuff is affected.

Same as the need for a typedef for a C++ function became only necessary with supporting functionPointerValue (something I am a little critical of anyway, because a function that's not void(void) still needs to be cast).